module/prover.go

```diff
@@ -134,8 +134,52 @@ func (pr *Prover) ProveHostConsensusState(ctx core.QueryContext, height exported
 
 // CheckRefreshRequired implements Prover.CheckRefreshRequired
 func (pr *Prover) CheckRefreshRequired(counterparty core.ChainInfoICS02Querier) (bool, error) {
-	// TODO implement
-	return false, nil
+	cpQueryHeight, err := counterparty.LatestHeight()
+	if err != nil {
+		return false, fmt.Errorf("failed to get the latest height of the counterparty chain: %v", err)
+	}
+	cpQueryCtx := core.NewQueryContext(context.TODO(), cpQueryHeight)
+
+	resCs, err := counterparty.QueryClientState(cpQueryCtx)
+	if err != nil {
+		return false, fmt.Errorf("failed to query the client state on the counterparty chain: %v", err)
+	}
+
+	var cs exported.ClientState
+	if err := pr.chain.Codec().UnpackAny(resCs.ClientState, &cs); err != nil {
+		return false, fmt.Errorf("failed to unpack Any into tendermint client state: %v", err)
+	}
+
+	resCons, err := counterparty.QueryClientConsensusState(cpQueryCtx, cs.GetLatestHeight())
+	if err != nil {
+		return false, fmt.Errorf("failed to query the consensus state on the counterparty chain: %v", err)
+	}
+
+	var cons exported.ConsensusState
+	if err := pr.chain.Codec().UnpackAny(resCons.ConsensusState, &cons); err != nil {
+		return false, fmt.Errorf("failed to unpack Any into tendermint consensus state: %v", err)
+	}
+	lcLastTimestamp := time.Unix(int64(cons.GetTimestamp()), 0)
```

Comment:
please don't forget to fix this line after fixing https://github.com/datachainlab/besu-ibc-relay-prover/pull/12/files#r1912604468